nuget/spec/dependabot/nuget/version_spec.rb

```diff
@@ -120,6 +136,10 @@
       end
     end
 
+    it "sorts versions correctly" do
```

Comment:
This test is failing because the current implementation does not sort versions correctly.
```
expected: ["", "0.9.0", "1.0.0-alpha", "1.0.0-alpha.1", "1.0.0-alpha.beta", "1.0.0-beta", "1.0.0-beta.2", "1.0.0-beta.11", "1.0.0-beta.11.1", "1.0.0-beta-extra-hyphens", "1.0.0-rc.1", "1.0.0", "1.0.0.1-pre", "1.0.0.1-pre.2", "1.0.0.1", "1.0.0.2", "1.1.0"]

got: ["", "0.9.0", "1.0.0", "1.0.0-alpha", "1.0.0-alpha.1", "1.0.0-alpha.beta", "1.0.0-beta", "1.0.0-beta-extra-hyphens", "1.0.0-beta.11", "1.0.0-beta.11.1", "1.0.0-beta.2", "1.0.0-rc.1", "1.0.0.1", "1.0.0.1-pre", "1.0.0.1-pre.2", "1.0.0.2", "1.1.0"]
```

Reply:
I ran these versions through the official nuget version class and sorted them and the testcase you added is correct, with the exception that an empty string isn't a valid nuget version, so that will have to be removed.

Reply:
Looking at this further, the test a few lines up does the actual version compare and seems to be correct; the test here appears to be doing a string sort.

Reply:
You're right! I wrote a bad test for this. I've tested with an updated test and all is well.